default.nix

```diff
@@ -0,0 +1,81 @@
+# To include benchmark deps use:
+# nix-shell --argstr c2nix "--benchmark"
+#
+# Use cabal flags as needed, e.g. to include examples:
+# nix-shell --argstr c2nix "--flag examples"
+#
+# To build the chart executable for running bench.sh use:
+# nix-shell --argstr c2nix "--flag dev" --run "cabal build chart"
+{
+  nixpkgs ?
+    import (builtins.fetchTarball https://github.com/composewell/nixpkgs/archive/f0ec167983f.tar.gz)
+        # fusion-plugin is marked a broken
+        { config.allowBroken = true;}
+, compiler ? "default"
+, c2nix ? "" # cabal2nix CLI options
+}:
+let haskellPackages =
+        if compiler == "default"
+        then nixpkgs.haskellPackages
+        else nixpkgs.haskell.packages.${compiler};
+
+    # we can possibly avoid adding our package to HaskellPackages like
+    # in the case of nix-shell for a single package?
+    mkPackage = self: pkg: path: opts: inShell:
+                let orig = self.callCabal2nixWithOptions pkg path opts {};
+                 in if inShell
+                    # Avoid copying the source directory to nix store by using
+                    # src = null.
+                    then orig.overrideAttrs (oldAttrs: { src = null; })
+                    else orig;
+
+    mkHaskellPackages = inShell:
+        haskellPackages.override {
+            overrides = self: super:
+                with nixpkgs.haskell.lib;
+                {
+                    streamly = mkPackage self "streamly" ./. c2nix inShell;
+                    streamly-benchmarks =
+                        mkPackage self "streamly-benchmarks"
+                            ./benchmark c2nix inShell;
+                    QuickCheck = self.QuickCheck_2_14;
+
```

Comment:
You might need to add a `splitmix = haskellPackages.splitmix_0_1_0_1;`, a depedency of `QuickCheck_2_14`

`self` here would cause an infinite recursion as `haskellPackages.callPackage` depends on `splitmix`

Reply:
Its a working/tested code. Note I am using      `import (builtins.fetchTarball https://github.com/composewell/nixpkgs/archive/f0ec167983f.tar.gz)`.

Reply:
Ok